subsys/bluetooth/host/scan.c

```diff
@@ -805,6 +805,14 @@ void bt_hci_le_per_adv_report(struct net_buf *buf)
 			 */
 			bt_hci_le_per_adv_report_recv(per_adv_sync, &buf->b, &info);
 		} else {
+			if (evt->data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_INCOMPLETE) {
+				LOG_WRN("Received incomplete advertising data. "
+					"Advertising report dropped.");
+
+				per_adv_sync->report_truncated = true;
+				net_buf_simple_reset(&per_adv_sync->reassembly);
+				return;
+			}
```

Comment:
Given that we `return` here, shouldn't this block be put into an `else if`, instead of the current approach?

Reply:
Yeah I wondered that as well, but did it this way since it was done the same way in [the lines below it ](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/scan.c#L808-L817). But can move it to its own if else if that is preferred.

Reply:
I would prefer an `else if` here. 

It is a bit tricky given the existing structure, but ideally it should have been
```c
if (evt->status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE) {
    /* do stuff */
} else if (evt->status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_INCOMPLETE) {
    /* do stuff */
} else if (evt->status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE) {
   /* do stuff */ 
}
```

Reply:
I've restructured the function a bit to follow the structure you suggested. Let me know if it looks ok.